Update node dependencies

jonathanthiry opened the pull request:
It looks like [rc-datepicker](https://github.com/buildo/rc-datepicker) is not compatible with React 17 and that it's not maintained anymore. I'm suggesting to use [react-datepicker](https://github.com/Hacker0x01/react-datepicker) instead. You can try it in the portal app by using this branch, it looks like this:

<img width="414" alt="Screenshot 2021-12-23 at 16 30 37" src="https://user-images.githubusercontent.com/991720/147261483-898c3a78-7044-41da-893a-2a737e1db2ec.png">

I updated the portal app for now and plan to do the other apps next.

mirzov commented on src/main/js/stats/main/components/PickDates.jsx:
```diff
@@ -1,7 +1,6 @@
 import React, { Component } from 'react';
 import ReactDOM from 'react-dom';
 import { DatePickerInput } from 'rc-datepicker';
-import '../../node_modules/rc-datepicker/lib/style.css';
```
This file belongs to stats app, why change it?

mirzov reviewed (commented):
Looks good to me! I updated nodejs to version 16 and npm to version 8, and ran `npm ci` on my machine, and got the following warnings from npm:
```
npm WARN old lockfile 
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile 
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile 
npm WARN deprecated @types/classnames@2.3.1: This is a stub types definition. classnames provides its own type definitions, so you do not need this installed.
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated topojson@3.0.2: Use topojson-client, topojson-server or topojson-simplify directly.
npm WARN deprecated core-js@1.2.7: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
npm WARN deprecated core-js@2.6.12: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
```

Should we do something about those? Should we agree about what npm/nodejs version to run?

roger-groth commented:
Looks ok to me. Perhaps add showMonthDropdown and showYearDropdown (at least showYearDropdown) to make it easier to jump to a specific data further back.

jonathanthiry commented:
I updated all the dependencies for the stats app except for `react-widgets`. The latest version (5.5.1) is conflicting with another package and I cannot figure out the issue. The current version (4.6.1) still works fine, and the new version is working in the portal app. Please take a look if you feel like it.

I replaced `rc-datepicker` with `react-datepicker` like I did in the portal app. I'm also using dropdowns for the month and year in both apps, like Roger suggested.

<img width="391" alt="date picker dropdown" src="https://user-images.githubusercontent.com/991720/148065589-099db354-6c3c-4a61-80a7-54399cac998b.png">
